flocker/node/_model.py

```diff
@@ -189,6 +164,39 @@ class Application(object):
     """
 
 
+@attributes(["dataset", "primary"])
+class Manifestation(object):
+    """
+    A dataset that is mounted on a node.
+
+    :ivar Dataset: The dataset being mounted.
+
+    :ivar bool primary: If true, this is a primary, otherwise it is a replica.
+    """
+
+
+@attributes(["dataset_id",
+             Attribute("maximum_size", default_value=None),
+             Attribute("metadata", default_factory=pmap)])
+class Dataset(object):
+    """
+    The filesystem data for a particular application.
+
+    :ivar dataset_id: A unique identifier, as ``unicode``. May also be ``None``
+        if this is coming out of human-supplied configuration, in which
+        case it will need to be looked up from actual state for existing
+        datasets, or a new one generated if a new dataset will need tbe
+        created.
+
+    :ivar PMap metadata: Mapping between ``unicode`` keys and
+        corresponding values. Typically there will be a ``"name"`` key whose
+        value is a a human-readable name, e.g. ``"main-postgres"``.
+
+    :ivar int maximum_size: The maximum size in bytes of this dataset, or
+        ``None`` if there is no specified limit.
```

Comment:
Add an issue and link to it in a comment around here about how this should be metadata instead.